cert-tools/snapstore/src/snapstore/snaps.py

```diff
@@ -0,0 +1,47 @@
+import re
+from dataclasses import dataclass, astuple
+
+@dataclass(frozen=True)
+class SnapChannel:
+    track: str | None = None
+    risk: str | None = None
+    branch: str | None = None
+
+    def __post_init__(self):
+        if self.track is None and self.risk is None:
+            raise ValueError("At least one of track or risk must be set")
+
+    @classmethod
+    def from_string(cls, string):
+        channel_template = r'^(?:([\w.-]+)(?:/([\w-]+)(?:/([\w-]+))?)?)?$'
```

Comment:
I know we have tests to document the regex but can you add an inline comment with an example anyway?